platform/engineering/request-for-change/2020-01-08-ssoe-removing-implicit-mvi-queries.md

```diff
@@ -0,0 +1,106 @@
+# RFC: Removing implicit MVI lookups
+
+- Date: _2020-01-08_
+
+
+## Background
+At present, when a user authenticates and is determined to be Level of assurance 3, we look up that user's VA profile in MVI (aka MPI, used interchangeably hereafter). The combination of the incoming attributes from the identity provider and the attributes from MVI make up the bulk of the logical user model that is used across vets-api.
+
+As part of the overall SSO consolidation work, we are integrating VA SSOe with the VA.gov sign-in process. We will maintain the three current credential options, but the SAML request/response cycle will be intercepted by SSOe before being returned to VA.gov. Along the way, SSOe can establish a session cookie for use by other sites. But more importantly, SSOe will provide a standardized SAML payload that includes attributes from the credential provider AND from MVI. 
+
+Since we will be receiving the bulk of the MVI attributes in the SAML payload, we want to determine whether we can and should do away with the separate MVI query that normally happens immediately following authentication.
```

Comment:
This is certainly going to be a lot of work in redoing specs across the board, might be an opportunity to clean up some things with how we instantiate current_user in our request specs currently. Totally makes sense. Just thinking out load.

Reply:
That's a good point in terms of LOE to get this done.